File: src/lib/software/google_drive.py
from src.lib.software import BaseSoftware

class GoogleDrive(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'GoogleDriveSetup.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe')

File: packages/app/src/services/software/definitions/googleDrive.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'google-drive';
	public name = 'Google Drive';
	public category = [SoftwareCategory.FileManagement, SoftwareCategory.Utility];
	public downloadName = 'GoogleDriveSetup.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = true;
	public deprecated = false;
	public icon = 'google-drive.png';
	public homepage = 'https://workspace.google.com/products/drive';

	public async resolveDownloadUrl() {
		return ok('https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe');
	}
}
